tests/virtualization/livemigration/live_migration_dst.pm

```diff
@@ -0,0 +1,62 @@
+# Copyright © 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Live migration test for destination host
+#
+# - Wait for guest migration initiated from source host after installation.
+#
+# Maintainer: Tony Yuan <tyuan@suse.com>
+
+package live_migration_dst;
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use Utils::Systemd 'systemctl';
+use lockapi;
+use mmapi;
+use virt_autotest::utils 'collect_virt_system_logs';
+require 'sle/tests/virtualization/livemigration/live_migration_src.pm';
+
+my $images_path = "/var/lib/libvirt/images";
+
+barrier_create('let_dst_upload_logs', 2);
+
+sub run {
+    select_console 'root-ssh';
+    # live_migration_src::cleanup_for_rerun() if (check_var('HOST_INSTALL', '0'));
+    if (check_var('HOST_INSTALL', 0)) {
+        live_migration_src::cleanup_for_rerun();
+    } else {
+        upload_logs('/var/log/zypp/history');
+    }
+
+    #Wait for the guests to be created and ready for migration
+    mutex_create 'dst_ready';
+    #wait to collect logs
+    barrier_wait {name => 'let_dst_upload_logs', check_dead_job => 1};
+    #Logs won't be collected if test is passed on src host
+    if (script_run("[[ -f $images_path/test_state/test_done ]]") == 0) {
+        assert_script_run("dmesg --level=emerg,crit,alert,err > /tmp/dmesg_err.txt");
+        upload_logs('/tmp/dmesg_err.txt') if (script_run("[[ -s  /tmp/dmesg_err.txt ]]") == 0);
+        assert_script_run("rm $images_path/test_state/test_done");
```

Comment:
Shall above uploaded logs also be uploaded when test_done is not there? Although collect_virt_system_logs will upload dmesg logs, but not filtered.

Reply:
@tonyyuan1 The above log upload for dmesg err log seems to repeat in this PR for several times, can it be wrapped in a function?

Reply:
test_done and collect_virt_system_logs is explained with  barrier_create/barrier_wait below.

In "dmesg --level=emerg,crit,alert,err" the "emerg,crit,alert,err" are filters. This dmesg out: dmesg_err.txt will be uploaded if there are messages in it. Will not be uploaded if there is no messages in it.
If the migration test fails it will not be uploaded either as /var/log/messages will be uploaded, which contains all log of it.

I thought about a function. I gaved up as there are just two lines and used twice.

Reply:
No big deal, fine.